litellm/llms/huggingface/chat/transformation.py

```diff
@@ -209,381 +97,67 @@ def transform_request(
-                    ),
-                    final_prompt_value=model_prompt_details.get(
-                        "final_prompt_value", ""
-                    ),
-                    bos_token=model_prompt_details.get("bos_token", ""),
-                    eos_token=model_prompt_details.get("eos_token", ""),
-                    messages=messages,
-                )
-            else:
-                prompt = prompt_factory(model=model, messages=messages)
-            inference_params = deepcopy(optional_params)
-            inference_params.pop("details")
-            inference_params.pop("return_full_text")
-            data = {
-                "inputs": prompt,  # type: ignore
-            }
-            if task == "text-generation-inference":
-                data["parameters"] = inference_params
-                data["stream"] = (  # type: ignore
-                    True  # type: ignore
-                    if "stream" in optional_params and optional_params["stream"] is True
-                    else False
-                )
-
-        ### RE-ADD SPECIAL PARAMS
-        if len(special_params_dict.keys()) > 0:
-            data.update({"options": special_params_dict})
-
-        return data
-
-    def get_api_base(self, api_base: Optional[str], model: str) -> str:
-        """
-        Get the API base for the Huggingface API.
-
-        Do not add the chat/embedding/rerank extension here. Let the handler do this.
-        """
-        if "https" in model:
-            completion_url = model
-        elif api_base is not None:
-            completion_url = api_base
-        elif "HF_API_BASE" in os.environ:
-            completion_url = os.getenv("HF_API_BASE", "")
-        elif "HUGGINGFACE_API_BASE" in os.environ:
-            completion_url = os.getenv("HUGGINGFACE_API_BASE", "")
-        else:
-            completion_url = f"https://api-inference.huggingface.co/models/{model}"
-
-        return completion_url
-
-    def validate_environment(
-        self,
-        headers: Dict,
-        model: str,
-        messages: List[AllMessageValues],
-        optional_params: Dict,
-        api_key: Optional[str] = None,
-        api_base: Optional[str] = None,
-    ) -> Dict:
-        default_headers = {
-            "content-type": "application/json",
-        }
-        if api_key is not None:
-            default_headers["Authorization"] = (
-                f"Bearer {api_key}"  # Huggingface Inference Endpoint default is to accept bearer tokens
-            )
-
-        headers = {**headers, **default_headers}
-        return headers
-
-    def get_error_class(
-        self, error_message: str, status_code: int, headers: Union[dict, httpx.Headers]
-    ) -> BaseLLMException:
-        return HuggingfaceError(
-            status_code=status_code, message=error_message, headers=headers
-        )
+        if "max_retries" in optional_params:
+            optional_params.pop("max_retries", None)
```

Comment:
maybe add a warning if `max_retries` is ignored? 

```suggestion
        if optional_params.pop("max_retries", None):
            warnings.warn(...) # "ignoring max_retries value"
```
(=> no need for `if "max_retries" in optional_params:` anymore)